rackcity/views/change_plan_views.py

```diff
@@ -0,0 +1,294 @@
+from django.http import JsonResponse
+from rackcity.api.serializers import AddChangePlanSerializer, GetChangePlanSerializer
+from rackcity.models import ChangePlan
+from rackcity.views.rackcity_utils import get_filter_arguments, get_sort_arguments
+from rackcity.utils.errors_utils import (
+    Status,
+    GenericFailure,
+    parse_serializer_errors,
+    parse_save_validation_error)
+from http import HTTPStatus
+from rest_framework.decorators import permission_classes, api_view
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.pagination import PageNumberPagination
+from rest_framework.parsers import JSONParser
+from django.core.exceptions import ObjectDoesNotExist
+import math
+
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def change_plan_delete(request):
+    """
+    Delete a single existing change plan
+    """
+    data = JSONParser().parse(request)
+    if 'id' not in data:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.DELETE_ERROR.value + GenericFailure.INTERNAL.value,
+                "errors": "Must include 'id' when deleting an asset"
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    id = data['id']
+    try:
+        existing_change_plan = ChangePlan.objects.get(id=id)
+
+    except ObjectDoesNotExist:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.MODIFY_ERROR.value +
+                    "Model" + GenericFailure.DOES_NOT_EXIST.value,
+                "errors": "No existing asset with id="+str(id)
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    try:
+        existing_change_plan.delete()
+    except Exception as error:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.DELETE_ERROR.value +
+                    "Change Plan" +
+                    GenericFailure.ON_DELETE.value,
+                "errors": str(error)
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    return JsonResponse(
+        {
+            "success_message":
+                Status.SUCCESS.value +
+                "Change Plan " + str(existing_change_plan.name) + " deleted"
```

Comment:
it's already deleted at this point, so calling `.name` will throw an error. you'll need to save the name to a variable before you delete it and then stringify that here

Reply:
ive tested deleting and it doesn't seem to mind?

Reply:
oh lit